pkg/shell/topnav.tsx

```diff
@@ -52,14 +72,14 @@ export class TopNav extends React.Component {
             theme: localStorage.getItem('shell:style') || 'auto',
         };
 
-        this.superuser_connection = null;
-        this.superuser = null;
-
         read_os_release().then(os => this.setState({ osRelease: os || {} }));
-
-        this.handleClickOutside = () => this.setState({ menuOpened: false, docsOpened: false });
     }
 
+    superuser_connection: cockpit.DBusClient | null = null;
+    superuser: cockpit.DBusProxy | null = null;
+
+    handleClickOutside = () => this.setState({ menuOpened: false, docsOpened: false });
```

Comment:
Defining this outside of the ctor at the class level looks weird. What would `this` even be here? I suppose the `this` gets resolved at call time even for an arrow, but then wouldn't that be the `this` of the call environment instead of the TopNav instance?

The tests prove it works -- my eyes just trip over it. JS has too much magic!

Reply:
> Defining this outside of the ctor at the class level looks weird. 

I guess... I moved it back into the constructor.